### PR TITLE
feat: order course blocks dictionary

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR migrations the most_recent_course_blocks_mv from Alembic to DBT.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by Ian2012** [comment]
@SoryRawyer I was trying to get the course structure sorted however, the `order by` is what defines how to replace in the `ReplacingMergeTree` engine. Do you have any ideas on how to solve it?

Should I create another model with only the data sorted and make the dictionaries query from it?

File: models/base/most_recent_course_blocks.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{{
+    config(
+        materialized="materialized_view",
+        schema=env_var("ASPECTS_EVENT_SINK_DATABASE", "event_sink"),
+        engine=get_engine("ReplacingMergeTree()"),
+        primary_key=("location"),
+        order_by=("location"),
+        post_hook="OPTIMIZE TABLE {{ this }} {{ on_cluster() }} FINAL",
```
Comment (SoryRawyer) [inline models/base/most_recent_course_blocks.sql:8]: I think we might be able to get rid of both the `order_by` and `post_hook` arguments.

**Review by SoryRawyer — COMMENTED** [review]
I like this idea, but since this is involved in creating `course_block_names_dict` in alembic (which isn't ready to move until after we add dictionary support to dbt-clickhouse) I think this might interfere with running the alembic migrations. @bmtcril what do you think?

**Comment by bmtcril** [comment]
Ah right, when I looked through this before I didn't see this line, but until we can move the dictionaries this needs to stay in alembic: https://github.com/openedx/tutor-contrib-aspects/blob/d2d9b88c5d9f9f6f9bc38fc14505735e83b7d8b4/tutoraspects/templates/aspects/apps/aspects/migrations/alembic/versions/0023_extend_display_names.py#L137

**Comment by Ian2012** [comment]
Closing it for now

**Comment by openedx-webhooks** [comment]
@Ian2012 Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/aspects-dbt&entry.867055334=https://github.com/openedx/aspects-dbt/pull/50&entry.1484655318=https://github.com/Ian2012&entry.752974735=2024-02-28+18:52&entry.1917517419=2024-02-29+17:28&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**